synapse/handlers/_base.py

```diff
@@ -208,7 +207,18 @@ def _create_new_client_event(self, builder):
             prev_member_event = yield self.store.get_room_member(
                 builder.sender, builder.room_id
             )
-            if prev_member_event:
+
+            # if we have the prev_member_event in context, we didn't receive
+            # the invite over federation. (More likely is that the inviting
+            # user, and all other local users, have left, making
+            # is_host_in_room return false).
+            #
+            context_events = (e.event_id for e in context.current_state.values())
```

Comment:
I'm a bit confused about this. The canonical check for if we received the invite over federation is if `inviter.domain != self.server_name`?